lib/ui/semantics.dart

```diff
@@ -742,22 +857,39 @@ class SemanticsUpdateBuilder extends NativeFieldWrapperClass2 {
     required double elevation,
     required double thickness,
     required Rect rect,
-    required String label,
-    required String hint,
-    required String value,
-    required String increasedValue,
-    required String decreasedValue,
+    // TODO(chunhtai): removes non attributed string parameters when framework pr is merged.
```

Comment:
```suggestion
    // TODO(chunhtai): remove non attributed string parameters when framework pr is merged.
```

Why will it be better to forbid non-attributed strings? Attributed strings require more time and space to construct and use.  Unless perhaps we optimize them so that if no attributes are specified no native work is done...

Reply:
I forbid it to make the api cleaner. If the performance is a concern, the other way will be making this interface to take string and attributes separately.

Reply:
Or just have a fast-path if the attributes list is empty, which could be exposed via some private field on the dart object.

Reply:
Is it possible to fast path it? The AttributedString is a native wrapper. When it reach the engine side, it becomes the peer object. If we don't set anything on the peer object when there is no attributes, how do we get the string from the peer object? Unless we receives it as a dart handle on the engine side, and unwrap it ourselves (I assume there is a way to do this?). I am not sure which is better

Reply:
YEah it's probably not worth conditionally creating the native peer or not. But I'm curious about how it'll affect a11y performance if you have to create additional native peers for every string that gets passed to native.

That said, I'm still wondering how this API compares to APIs developers are used to here - although this lower level API so it's probably fine, developers won't be interacting with this much.

Reply:
Only people that write a UI framework on top of the engine will probably interact with this API directly...

Reply:
The string would need to be passed to the engine side anyway, I think the only part that is problematic is that user create a bunch of temporary AttributedStrings. The ImageFilter has the same issue I think?

Another way to do this is moving the AttributedString to the framework side, so only the StringAttribute will create peer. and makes this interface to take in string and list of attributes. one problem with this approach tho is I will need to expose StringAttribute.copy() because now they are in different libraries.

Reply:
Do we know what the performance penalty is for creating the native objects? If that's too much, maybe we should just keep (and use) the non-attributed versions where appropriate?

Reply:
I think I will move the AttributedString to the framework side and make the interface take string and attributes list separately. This way creating attributedString without StringAtrribtue will not create any native object.

I against the idea of separating attributed version and non attributed version in the framework side. It will be very hard to maintain separate code paths when doing assert and update check. for example how do we merge semantics nodes that have different combinations of attributedString and non attributed string.